gRPC server intermittently fails to be terminated when sync with people completes

So next time you try to sync with people, the gRPC server can't be started and the following exception is unhandled: `java.io.IOException: Failed to bind`

I've run into this a few times whilst testing, but I haven't figured out how to reproduce it consistently.

Here's the stacktrace:

```
2020-05-13 16:34:04.766 6190-6190/? I/zygote64: Late-enabling -Xcheck:jni
2020-05-13 16:34:04.927 6190-6190/tech.relaycorp.courier W/System.err: java.io.FileNotFoundException: /jacoco.exec (Read-only file system)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at java.io.FileOutputStream.open0(Native Method)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at java.io.FileOutputStream.open(FileOutputStream.java:287)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at java.io.FileOutputStream.<init>(FileOutputStream.java:223)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.openFile(FileOutput.java:67)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.output.FileOutput.startup(FileOutput.java:49)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Agent.startup(Agent.java:122)
2020-05-13 16:34:04.928 6190-6190/tech.relaycorp.courier W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Agent.getInstance(Agent.java:50)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Offline.<clinit>(Offline.java:31)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at org.jacoco.agent.rt.internal_8ff85ea.Offline.getProbes(Offline.java:51)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at tech.relaycorp.courier.App.$jacocoInit(Unknown Source:13)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at tech.relaycorp.courier.App.<init>(Unknown Source:0)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at java.lang.Class.newInstance(Native Method)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.Instrumentation.newApplication(Instrumentation.java:1102)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.Instrumentation.newApplication(Instrumentation.java:1087)
2020-05-13 16:34:04.929 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.LoadedApk.makeApplication(LoadedApk.java:983)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5715)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.ActivityThread.-wrap1(Unknown Source:0)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:106)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at android.os.Looper.loop(Looper.java:164)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:6494)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
2020-05-13 16:34:04.930 6190-6190/tech.relaycorp.courier W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
2020-05-13 16:34:05.006 6190-6196/tech.relaycorp.courier I/zygote64: Do partial code cache collection, code=20KB, data=29KB
2020-05-13 16:34:05.006 6190-6196/tech.relaycorp.courier I/zygote64: After code cache collection, code=20KB, data=29KB
2020-05-13 16:34:05.006 6190-6196/tech.relaycorp.courier I/zygote64: Increasing code cache capacity to 128KB
2020-05-13 16:34:05.026 6190-6196/tech.relaycorp.courier I/zygote64: Do partial code cache collection, code=24KB, data=58KB
2020-05-13 16:34:05.027 6190-6196/tech.relaycorp.courier I/zygote64: After code cache collection, code=24KB, data=58KB
2020-05-13 16:34:05.027 6190-6196/tech.relaycorp.courier I/zygote64: Increasing code cache capacity to 256KB
2020-05-13 16:34:05.093 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:34:05.111 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2436)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.BaseActivity.setContentView(int) (BaseActivity.kt:32)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.main.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:39)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle, android.os.PersistableBundle) (Activity.java:7009)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:7000)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1214)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2731)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Looper.loop() (Looper.java:164)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk"],nativeLibraryDirectories=[/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/lib/arm64, /data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2436)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.BaseActivity.setContentView(int) (BaseActivity.kt:32)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.main.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:39)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle, android.os.PersistableBundle) (Activity.java:7009)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:7000)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1214)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2731)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Looper.loop() (Looper.java:164)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
2020-05-13 16:34:05.145 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2436)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.BaseActivity.setContentView(int) (BaseActivity.kt:32)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.main.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:39)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle, android.os.PersistableBundle) (Activity.java:7009)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:7000)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1214)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2731)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Looper.loop() (Looper.java:164)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk"],nativeLibraryDirectories=[/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/lib/arm64, /data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2436)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.BaseActivity.setContentView(int) (BaseActivity.kt:32)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.main.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:39)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle, android.os.PersistableBundle) (Activity.java:7009)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:7000)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1214)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2731)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Looper.loop() (Looper.java:164)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64: Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2436)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.BaseActivity.setContentView(int) (BaseActivity.kt:32)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.main.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:39)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle, android.os.PersistableBundle) (Activity.java:7009)
2020-05-13 16:34:05.146 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:7000)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1214)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2731)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Looper.loop() (Looper.java:164)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk"],nativeLibraryDirectories=[/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/lib/arm64, /data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2436)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.BaseActivity.setContentView(int) (BaseActivity.kt:32)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void tech.relaycorp.courier.ui.main.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:39)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle, android.os.PersistableBundle) (Activity.java:7009)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:7000)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1214)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2731)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2856)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.-wrap11(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1589)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:106)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.os.Looper.loop() (Looper.java:164)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6494)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run() (RuntimeInit.java:438)
2020-05-13 16:34:05.147 6190-6190/tech.relaycorp.courier I/zygote64:     at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:807)
2020-05-13 16:34:05.459 6190-6196/tech.relaycorp.courier I/zygote64: Compiler allocated 7MB to compile void android.widget.TextView.<init>(android.content.Context, android.util.AttributeSet, int, int)
2020-05-13 16:34:05.630 6190-6253/tech.relaycorp.courier I/Adreno: QUALCOMM build                   : 8e59954, I0be83d0d26
    Build Date                       : 09/22/17
    OpenGL ES Shader Compiler Version: EV031.21.02.00
    Local Branch                     : O17A
    Remote Branch                    : 
    Remote Branch                    : 
    Reconstruct Branch               : 
2020-05-13 16:34:05.641 6190-6253/tech.relaycorp.courier I/Adreno: PFP: 0x00000000, ME: 0x00000000
2020-05-13 16:34:05.649 6190-6253/tech.relaycorp.courier I/zygote64: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
2020-05-13 16:34:05.649 6190-6253/tech.relaycorp.courier I/OpenGLRenderer: Initialized EGL, version 1.4
2020-05-13 16:34:05.729 6190-6196/tech.relaycorp.courier I/zygote64: Do full code cache collection, code=111KB, data=123KB
2020-05-13 16:34:05.729 6190-6196/tech.relaycorp.courier I/zygote64: After code cache collection, code=105KB, data=60KB
2020-05-13 16:34:05.787 6190-6196/tech.relaycorp.courier I/zygote64: Do partial code cache collection, code=124KB, data=74KB
2020-05-13 16:34:05.788 6190-6196/tech.relaycorp.courier I/zygote64: After code cache collection, code=124KB, data=74KB
2020-05-13 16:34:05.788 6190-6196/tech.relaycorp.courier I/zygote64: Increasing code cache capacity to 512KB
2020-05-13 16:34:54.524 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:34:54.529 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:34:54.670 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:34:54.675 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:34:55.308 6190-6190/tech.relaycorp.courier I/CogRPCServer: Server stopped
2020-05-13 16:35:06.903 6190-6190/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 12
2020-05-13 16:35:06.992 6190-6190/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 13
2020-05-13 16:35:07.246 6190-6190/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 10
2020-05-13 16:35:07.299 6190-6190/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 11
2020-05-13 16:35:10.166 6190-6190/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 12
2020-05-13 16:35:10.246 6190-6190/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 13
2020-05-13 16:35:13.998 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:35:14.002 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 16:35:14.060 6190-6196/tech.relaycorp.courier I/zygote64: Compiler allocated 4MB to compile void android.view.View.<init>(android.content.Context, android.util.AttributeSet, int, int)
2020-05-13 16:35:14.085 6190-6196/tech.relaycorp.courier I/zygote64: Do full code cache collection, code=251KB, data=166KB
2020-05-13 16:35:14.087 6190-6196/tech.relaycorp.courier I/zygote64: After code cache collection, code=225KB, data=114KB
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64: Rejecting re-init on previously-failed class java.lang.Class<io.netty.util.internal.logging.Log4J2Logger>: java.lang.NoClassDefFoundError: Failed resolution of: Lorg/apache/logging/log4j/spi/ExtendedLoggerWrapper;
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLogger io.netty.util.internal.logging.Log4J2LoggerFactory.newInstance(java.lang.String) (Log4J2LoggerFactory.java:33)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLoggerFactory io.netty.util.internal.logging.InternalLoggerFactory.newDefaultFactory(java.lang.String) (InternalLoggerFactory.java:49)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLoggerFactory io.netty.util.internal.logging.InternalLoggerFactory.getDefaultFactory() (InternalLoggerFactory.java:69)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLogger io.netty.util.internal.logging.InternalLoggerFactory.getInstance(java.lang.String) (InternalLoggerFactory.java:92)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLogger io.netty.util.internal.logging.InternalLoggerFactory.getInstance(java.lang.Class) (InternalLoggerFactory.java:85)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.internal.PlatformDependent.<clinit>() (PlatformDependent.java:80)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at byte[] io.netty.util.internal.PlatformDependent.allocateUninitializedArray(int) (PlatformDependent.java:281)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.AsciiString.<init>(java.lang.CharSequence, int, int) (AsciiString.java:223)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.AsciiString.<init>(java.lang.CharSequence) (AsciiString.java:210)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.AsciiString io.netty.util.AsciiString.cached(java.lang.String) (AsciiString.java:1401)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.AsciiString.<clinit>() (AsciiString.java:48)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.AsciiString io.netty.util.AsciiString.of(java.lang.CharSequence) (AsciiString.java:1391)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.grpc.netty.Utils.<clinit>() (Utils.java:72)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.grpc.netty.NettyServerBuilder.<clinit>() (NettyServerBuilder.java:78)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.grpc.netty.NettyServerBuilder io.grpc.netty.NettyServerBuilder.forAddress(java.net.SocketAddress) (NettyServerBuilder.java:126)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.cogrpc.server.CogRPCServer$start$3.invokeSuspend(java.lang.Object) (CogRPCServer.kt:53)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.cogrpc.server.CogRPCServer$start$3.invoke(java.lang.Object, java.lang.Object) (CogRPCServer.kt:-1)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(kotlinx.coroutines.internal.ScopeCoroutine, java.lang.Object, kotlin.jvm.functions.Function2) (Undispatched.kt:91)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(kotlin.coroutines.CoroutineContext, kotlin.jvm.functions.Function2, kotlin.coroutines.Continuation) (Builders.common.kt:154)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object kotlinx.coroutines.BuildersKt.withContext(kotlin.coroutines.CoroutineContext, kotlin.jvm.functions.Function2, kotlin.coroutines.Continuation) ((null):1)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.cogrpc.server.CogRPCServer.start(tech.relaycorp.cogrpc.server.CogRPCServer$Service, kotlin.jvm.functions.Function1, kotlin.coroutines.Continuation) (CogRPCServer.kt:47)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.courier.domain.PrivateSync.startSync(kotlin.coroutines.Continuation) (PrivateSync.kt:25)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.courier.ui.sync.people.PeopleSyncViewModel$1.invokeSuspend(java.lang.Object) (PeopleSyncViewModel.kt:46)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(java.lang.Object) (ContinuationImpl.kt:33)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.DispatchedTask.run() (DispatchedTask.kt:56)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(kotlinx.coroutines.scheduling.Task) (CoroutineScheduler.kt:571)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(kotlinx.coroutines.scheduling.Task) (CoroutineScheduler.kt:738)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker() (CoroutineScheduler.kt:678)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run() (CoroutineScheduler.kt:665)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "org.apache.logging.log4j.spi.ExtendedLoggerWrapper" on path: DexPathList[[zip file "/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk"],nativeLibraryDirectories=[/data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/lib/arm64, /data/app/tech.relaycorp.courier-bnZ5lFrMT7gERmUdTDqk5Q==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:125)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLogger io.netty.util.internal.logging.Log4J2LoggerFactory.newInstance(java.lang.String) (Log4J2LoggerFactory.java:33)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLoggerFactory io.netty.util.internal.logging.InternalLoggerFactory.newDefaultFactory(java.lang.String) (InternalLoggerFactory.java:49)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLoggerFactory io.netty.util.internal.logging.InternalLoggerFactory.getDefaultFactory() (InternalLoggerFactory.java:69)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLogger io.netty.util.internal.logging.InternalLoggerFactory.getInstance(java.lang.String) (InternalLoggerFactory.java:92)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.internal.logging.InternalLogger io.netty.util.internal.logging.InternalLoggerFactory.getInstance(java.lang.Class) (InternalLoggerFactory.java:85)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.internal.PlatformDependent.<clinit>() (PlatformDependent.java:80)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at byte[] io.netty.util.internal.PlatformDependent.allocateUninitializedArray(int) (PlatformDependent.java:281)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.AsciiString.<init>(java.lang.CharSequence, int, int) (AsciiString.java:223)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.AsciiString.<init>(java.lang.CharSequence) (AsciiString.java:210)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.AsciiString io.netty.util.AsciiString.cached(java.lang.String) (AsciiString.java:1401)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.netty.util.AsciiString.<clinit>() (AsciiString.java:48)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.netty.util.AsciiString io.netty.util.AsciiString.of(java.lang.CharSequence) (AsciiString.java:1391)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.grpc.netty.Utils.<clinit>() (Utils.java:72)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void io.grpc.netty.NettyServerBuilder.<clinit>() (NettyServerBuilder.java:78)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at io.grpc.netty.NettyServerBuilder io.grpc.netty.NettyServerBuilder.forAddress(java.net.SocketAddress) (NettyServerBuilder.java:126)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.cogrpc.server.CogRPCServer$start$3.invokeSuspend(java.lang.Object) (CogRPCServer.kt:53)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.cogrpc.server.CogRPCServer$start$3.invoke(java.lang.Object, java.lang.Object) (CogRPCServer.kt:-1)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(kotlinx.coroutines.internal.ScopeCoroutine, java.lang.Object, kotlin.jvm.functions.Function2) (Undispatched.kt:91)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(kotlin.coroutines.CoroutineContext, kotlin.jvm.functions.Function2, kotlin.coroutines.Continuation) (Builders.common.kt:154)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object kotlinx.coroutines.BuildersKt.withContext(kotlin.coroutines.CoroutineContext, kotlin.jvm.functions.Function2, kotlin.coroutines.Continuation) ((null):1)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.cogrpc.server.CogRPCServer.start(tech.relaycorp.cogrpc.server.CogRPCServer$Service, kotlin.jvm.functions.Function1, kotlin.coroutines.Continuation) (CogRPCServer.kt:47)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.courier.domain.PrivateSync.startSync(kotlin.coroutines.Continuation) (PrivateSync.kt:25)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at java.lang.Object tech.relaycorp.courier.ui.sync.people.PeopleSyncViewModel$1.invokeSuspend(java.lang.Object) (PeopleSyncViewModel.kt:46)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(java.lang.Object) (ContinuationImpl.kt:33)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.DispatchedTask.run() (DispatchedTask.kt:56)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(kotlinx.coroutines.scheduling.Task) (CoroutineScheduler.kt:571)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(kotlinx.coroutines.scheduling.Task) (CoroutineScheduler.kt:738)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker() (CoroutineScheduler.kt:678)
2020-05-13 16:35:14.667 6190-6246/tech.relaycorp.courier I/zygote64:     at void kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run() (CoroutineScheduler.kt:665)
2020-05-13 16:35:14.723 6190-6246/tech.relaycorp.courier I/PlatformDependent: Your platform does not provide complete low-level API for accessing direct buffers reliably. Unless explicitly requested, heap buffer will always be preferred to avoid potential system instability.
2020-05-13 16:35:14.962 6190-6196/tech.relaycorp.courier I/zygote64: Do partial code cache collection, code=249KB, data=150KB
2020-05-13 16:35:14.964 6190-6196/tech.relaycorp.courier I/zygote64: After code cache collection, code=249KB, data=150KB
2020-05-13 16:35:14.964 6190-6196/tech.relaycorp.courier I/zygote64: Increasing code cache capacity to 1024KB
2020-05-13 16:35:15.062 6190-6246/tech.relaycorp.courier I/UndispatchedCoroutine: Server started
2020-05-13 16:36:00.562 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next 85d654c8-4b2b-44d4-a833-4766375f6da6
2020-05-13 16:36:00.684 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.685 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next f88a1935-b63e-4fd0-8c9f-b85bb5b7b7e2
2020-05-13 16:36:00.706 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.707 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next fd487227-b45f-4eed-8f9d-46ae98f0afd5
2020-05-13 16:36:00.725 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.726 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next 0d174a31-3e1a-4f78-9786-3c1f2d52eb04
2020-05-13 16:36:00.739 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.739 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next 54a2d277-8683-4b66-ac51-a4e97e233472
2020-05-13 16:36:00.751 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.752 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next 0fd5fd10-9d7c-48bd-ba7a-fc418b2c6a46
2020-05-13 16:36:00.762 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.763 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next 1e6cec43-963c-4984-8767-e15daa97fddc
2020-05-13 16:36:00.774 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.774 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next b7f57c1d-f23f-4f08-b5e0-c88f6731c492
2020-05-13 16:36:00.784 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.785 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next 6ce21f56-9f92-4907-91b8-17cf925ae991
2020-05-13 16:36:00.795 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:36:00.795 6190-6190/tech.relaycorp.courier I/StandaloneCoroutine: deliverCargo next b6a10800-3a80-4a03-8e06-9e947dc90e2f
2020-05-13 16:36:00.805 6190-6190/tech.relaycorp.courier W/StoreMessage: Invalid cargo received: Creation date is in the future
2020-05-13 16:41:12.134 6190-6190/tech.relaycorp.courier I/CogRPCServer: Server stopped
2020-05-13 17:01:54.635 6190-6885/tech.relaycorp.courier W/nService$deliverCargo$1: deliverCargo error
    io.grpc.StatusRuntimeException: CANCELLED: cancelled before receiving half close
        at io.grpc.Status.asRuntimeException(Status.java:524)
        at io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onCancel(ServerCalls.java:275)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closedInternal(ServerCallImpl.java:353)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closed(ServerCallImpl.java:341)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1Closed.runInContext(ServerImpl.java:868)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
2020-05-13 17:01:56.035 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 17:01:56.040 6190-6190/tech.relaycorp.courier E/ResourceType: Style contains key with bad entry: 0x01010604
2020-05-13 17:01:56.731 6190-6244/tech.relaycorp.courier W/UndispatchedCoroutine: Could not start server
    java.io.IOException: Failed to bind
        at io.grpc.netty.NettyServer.start(NettyServer.java:250)
        at io.grpc.internal.ServerImpl.start(ServerImpl.java:184)
        at io.grpc.internal.ServerImpl.start(ServerImpl.java:90)
        at tech.relaycorp.cogrpc.server.CogRPCServer$start$3.invokeSuspend(CogRPCServer.kt:69)
        at tech.relaycorp.cogrpc.server.CogRPCServer$start$3.invoke(Unknown Source:14)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:154)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at tech.relaycorp.cogrpc.server.CogRPCServer.start(CogRPCServer.kt:47)
        at tech.relaycorp.courier.domain.PrivateSync.startSync(PrivateSync.kt:25)
        at tech.relaycorp.courier.ui.sync.people.PeopleSyncViewModel$1.invokeSuspend(PeopleSyncViewModel.kt:46)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
     Caused by: java.net.BindException: Address already in use
        at sun.nio.ch.Net.bind0(Native Method)
        at sun.nio.ch.Net.bind(Net.java:442)
        at sun.nio.ch.Net.bind(Net.java:434)
        at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:224)
        at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
        at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:136)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:550)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1334)
        at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:506)
        at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:491)
        at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:973)
        at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:248)
        at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:356)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:764)
```

I think the culprit could be related to `io.grpc.StatusRuntimeException: CANCELLED: cancelled before receiving half close`.

--- a/app/src/main/java/tech/relaycorp/cogrpc/server/CogRPCServer.kt
+++ b/app/src/main/java/tech/relaycorp/cogrpc/server/CogRPCServer.kt
@@ -77,7 +77,7 @@ internal constructor(
     }
 
     fun stop() {
-        // server?.shutdown()
+        server?.shutdown()
         server = null
 
         job.cancel()

